bioio/plugins.py

```diff
@@ -53,24 +59,105 @@ def add_plugin(pluginentry: PluginEntry) -> None:
+            f"`{distribution_name}`, instead found {len(matching_distribution_deps)}"
+        )
+
+    # Versions from importlib are formatted like "<dependency name> (<0.19,>=0.18)"
+    # so this regex needs to find the inner part of the parentheses to get the
+    # version specifications
+    version_specification_match = re.findall(r"\(.*?\)", matching_distribution_deps[0])
+
+    minimum_dependency_version = None
+    maximum_dependency_version = None
+    if version_specification_match:
+        # A distribution can specify a minimum and maximum (or force an exact) version
+        # so multiple versions need to be parsed here
+        version_specifications = version_specification_match[0].split(",")
+        for version_specification in version_specifications:
+            version = "".join(
+                [
+                    char
+                    for char in version_specification
+                    if char.isnumeric() or char == "."
+                ]
+            )
+
+            # Unfortunately not all versions are specified in full SemVer format
+            # and instead can be like "0.19" and the "semver" package does not
+            # handle these well so we can just fill in the missing .0
+            # to make 0.19 -> 0.19.0
+            while version.count(".") < 2:
+                version += ".0"
+
+            if "<" in version_specification or "==" in version_specification:
+                maximum_dependency_version = version
+
+            if ">" in version_specification or "==" in version_specification:
+                minimum_dependency_version = version
+
+    return minimum_dependency_version, maximum_dependency_version
+
+
 def get_plugins() -> List[PluginEntry]:
     plugins = entry_points(group="bioio.readers")
+    (
+        min_compatible_bioio_base_version,
+        _,
+    ) = get_dependency_version_range_for_distribution(
+        BIOIO_DIST_NAME, BIOIO_BASE_DIST_NAME
+    )
     for plugin in plugins:
-        # ReaderMetadata knows how to instantiate the actual Reader
-        reader_meta = plugin.load().ReaderMetadata
-        if plugin.dist is not None:
-            files = plugin.dist.files
-            if files is not None:
-                firstfile = files[0]
-                timestamp = os.path.getmtime(Path(firstfile.locate()).parent)
-            else:
-                print(f"No files found for plugin: '{plugin}'")
+        (
+            _,
+            max_bioio_base_version_for_plugin,
+        ) = get_dependency_version_range_for_distribution(
+            plugin.name, BIOIO_BASE_DIST_NAME
+        )
+        if (
+            min_compatible_bioio_base_version is not None
+            and max_bioio_base_version_for_plugin is not None
+            and semver.compare(
+                max_bioio_base_version_for_plugin,
+                min_compatible_bioio_base_version,
+            )
+            < 0
+        ):
+            print(
```

Comment:
at some point these prints should maybe be logs instead; not necessarily for this pr